### PR TITLE
Bump vispy 0.11.0

### PR DESCRIPTION
**Pull request by brisvag**
# Description
Bump vispy to 0.11.0. This fixes/improves a few things:
- makes `Volume` able to use the same interpolation methods as `Image`. Beware: performance degrades *much* faster than in 2D when using bigger kernels :P
- fixes a bug where `blend_equations` were going "out of sync" (moves the work from @psobolewskiPhD on inverted LUTs forward)
- fixes the messed up clipping planes: #4503

The above are accounted for already in this PR. Additionally (but will need to be exposed in napari on a later PR):
- introduces the `custom` interpolation for images, which allows providing an arbitrary sampling kernel to be passed to the GPU (vispy/vispy#2319)

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Comment by Czaki** [comment]
I cannot agree that it is a nonbreaking change. I would like to see code that will handle `bilinear` and replace it with `linear` with an emitting warning.

**Comment by brisvag** [comment]
Actually, `bilinear` is still allowed (it was there for too long to break it), so it's probably better to allow it from code if one insists (no need for a deprecation, it won't get removed from vispy anyways), and simply not expose it from gui anymore. I just readded it to the options.

**Comment by brisvag** [comment]
Actually, I'm wrong; `bilinear` will not work for `Volume`, which doesn't have the fallback... need to deprecate it, then.

File: napari/_qt/layer_controls/qt_image_controls.py [hunk 1]
```diff
@@ -336,11 +335,7 @@ def _toggle_plane_parameter_visibility(self):
             self.planeThicknessLabel.show()
 
     def _update_interpolation_combo(self):
-        interp_names = (
-            Interpolation3D.keys()
-            if self.layer._ndisplay == 3
-            else [i.value for i in Interpolation.view_subset()]
-        )
+        interp_names = [i.value for i in Interpolation.view_subset()]
```
Comment (Czaki) [inline napari/_qt/layer_controls/qt_image_controls.py:338]: I'm not sure, but You told earlier that some rendering types are slow in 3D. Did I correct? If yes, then did we want to expose it to GUI?

**Review by Czaki — APPROVED** [review]
In the exception of `volume.py` (which I do not understand), I could confirm that everything is ok for me.

**Comment by brisvag** [comment]
> the exception of `volume.py` (which I do not understand)

Those changes were needed because that shader function was renamed in vispy; they are not doing anything, other than updating our shader so it doesn't break!

File: napari/_qt/layer_controls/qt_image_controls.py (hunk 1 above)
Comment (brisvag) [reply napari/_qt/layer_controls/qt_image_controls.py:338]: I'm up for anything here. We can probably hide from gui the really big ones (lanczos) but leave things like `bicubic`.

**Comment by Czaki** [comment]
> Those changes were needed because that shader function was renamed in vispy; they are not doing anything, other than updating our shader so it doesn't break!

I know, but I do not have enough experience to validate it.

File: napari/_qt/layer_controls/qt_image_controls.py (hunk 1 above)
Comment (andy-sweet) [reply napari/_qt/layer_controls/qt_image_controls.py:338]: As long as the default is something fast, I'm OK with just leaving them all in. For some users 3D performance might be good enough for various reasons (e.g. their data is small or they have a really fast graphics card).

File: napari/_vispy/visuals/volume.py [hunk 2]
```diff
@@ -35,7 +35,7 @@
 vec4 calculateCategoricalColor(vec4 betterColor, vec3 loc, vec3 step)
 {
     // Calculate color by incorporating ambient and diffuse lighting
-    vec4 color0 = $sample(u_volumetex, loc);
+    vec4 color0 = $get_data(loc);
```
Comment (andy-sweet) [inline napari/_vispy/visuals/volume.py:38]: I don't fully understand these changes, but I want to learn! I understand that these changes make our custom shader consistent with the changes in https://github.com/vispy/vispy/pull/2322

What was `sample` doing previously? Looks like it was an alias to the texture's `glsl_sample` method. I'm guessing that was handling 3d interp previously? And now we're just getting the color data and it's being handled elsewhere in the rendering pipeline (e.g. another shader)?

**Review by andy-sweet — COMMENTED** [review]
LGTM! Like @Czaki I don't fully understand the changes in `volume.py`, but I did ask a clarifying question. Maybe @jni will be better placed to approve those changes.

File: napari/_vispy/visuals/volume.py (hunk 2 above)
Comment (brisvag) [reply napari/_vispy/visuals/volume.py:38]: `$sample` was basically calling the hardware texture lookup. The new `$get_data` is named to reflect the equivalent in `ImageVisual`, and effectively calls the shader code inside `_INTERPOLATION_TEMPLATE` with whatever interpolation function is chosen (the actual shader for the intepolation lives in https://github.com/vispy/vispy/blob/main/vispy/glsl/misc/spatial-filters.frag).

Comment (andy-sweet) [reply napari/_vispy/visuals/volume.py:38]: Thanks for the explanation!

File: napari/_qt/layer_controls/qt_image_controls.py (hunk 1 above)
Comment (brisvag) [reply napari/_qt/layer_controls/qt_image_controls.py:338]: The risk of that is that a user might unkowingly pick a *very* slow one and grind napari to a halt :P

Comment (brisvag) [reply napari/_qt/layer_controls/qt_image_controls.py:338]: Though I guess for that we could learn something from #4764 and roll back to something sane if the FPS drop too much...

**Review by sofroniewn — APPROVED** [review]
This is great - I love that we can drop `Interpolation3D`! I will let @brisvag merge when he feels ready, but this can go in now I think

**Pull request merged**